### ISSUE TITLE
Krane is unusable on Ruby 2.7.0

### ISSUE THREAD

**Issue by benlangfeld** [Shopify/krane#690]
# Bug report

The upstream bug is this: https://github.com/aetherknight/recursive-open-struct/issues/62

### Steps to Reproduce

1. Run the krane tests on Ruby 2.7.0

**Comment by dturn** [comment]
We've had a least 1 report that its useable with lots of warnings on 2.7, https://github.com/Shopify/krane/pull/675. 

 I wonder if you can fix a gem version to make it work.

**Comment by benlangfeld** [comment]
Upstream fix is proposed @ https://github.com/aetherknight/recursive-open-struct/pull/64

**Comment by dturn** [comment]
Thanks for PRing this upstream!

**Comment by breunigs** [comment]
Upstream has released 1.1.1 which fixes these issues:
https://github.com/aetherknight/recursive-open-struct/pull/64#issuecomment-597469281

I've opened a PR against kubeclient to have them bump the min required version to ensure compatibility: https://github.com/abonas/kubeclient/pull/439 . Since krane doesn't directly depend on recursive-open-struct, it needs to wait for kubeclient to release a new version.

However, since krane doesn't ship with a `Gemfile.lock` it will automatically pull in the newest recursive-open-struct. Using it, the test suite passes for me locally, even when using the release version of kubeclient. Would you kindly restart the CI on https://github.com/Shopify/krane/pull/675 , because I assume it will pass now.

I will also provide a PR fixing more of these deprecation warnings.

**Comment by dturn** [comment]
CI is still failing for #675 . #710 is passing for ruby 2.6, I need to check the linting and ruby 2.7 but its promising.

**Comment by breunigs** [comment]
If you copy&paste the errors I can take a look, otherwise it's a bit much guess work :)

**Comment by cben** [comment]
Hi, coming here from https://github.com/abonas/kubeclient/pull/439.
My general understanding of semver patch releases, like recursive-open-struct 1.1.0 -> 1.1.1, is that few people _prevent_ them in dependency ranges, so once a fix is out there is no need to "bubble it up" by releasing new versions of all intermediate consumers.

If I'm wrong, please comment on https://github.com/abonas/kubeclient/pull/439.

Options I see from your perspective:
1. Do nothing.
2. Require `'recursive-open-struct', '~> 1.1', '>= 1.1.1'` in _your_ krane.gemspec.
3. Require it in kubeclient, I release new kubeclient, you bump kubeclient dependency.

Are there situations where (1) results in broken installs that (2) or (3) fix?

**Comment by dturn** [comment]
@cben 

I think (1) & (3) are both valid options. We'd prefer not to add transitive dependencies to our gemspec. 

When I run our tests I do get depreciation warnings coming from kubeclient, perhaps when you fix those you could bump recursive-open-struct?
```
/Users/danielturner/.gem/ruby/2.7.0/gems/kubeclient-4.6.0/lib/kubeclient.rb:27: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/Users/danielturner/.gem/ruby/2.7.0/gems/kubeclient-4.6.0/lib/kubeclient/common.rb:61: warning: The called method `initialize_client' is defined here
```

**Comment by dturn** [comment]
#710 has merged and which includes the needed fixes to make krane work with ruby 2.7

**Comment by dturn** [comment]
released in v1.1.3
